certbot-postfix/certbot_postfix/util.py

```diff
@@ -0,0 +1,285 @@
+
+        If Postfix isn't currently running, this method will fail.
+
+        :raises .PluginError: when Postfix cannot reload
+        """
+        try:
+            self._call(["reload"])
+        except subprocess.CalledProcessError:
+            raise errors.PluginError(
+                "Postfix failed to reload its configuration")
+
+def check_all_output(*args, **kwargs):
+    """A version of subprocess.check_output that also captures stderr.
+
+    This is the same as :func:`subprocess.check_output` except output
+    written to stderr is also captured and returned to the caller. The
+    return value is a tuple of two strings (rather than byte strings).
+    To accomplish this, the caller cannot set the stdout, stderr, or
+    universal_newlines parameters to :class:`subprocess.Popen`.
+
+    Additionally, if the command exits with a nonzero status, output is
+    not included in the raised :class:`subprocess.CalledProcessError`
+    because Python 2.6 does not support this. Instead, the failure
+    including the output is logged.
+
+    :param tuple args: positional arguments for Popen
+    :param dict kwargs: keyword arguments for Popen
+
+    :returns: data written to stdout and stderr
+    :rtype: `tuple` of `str`
+
+    :raises ValueError: if arguments are invalid
+    :raises subprocess.CalledProcessError: if the command fails
+
+    """
+    for keyword in ('stdout', 'stderr', 'universal_newlines',):
+        if keyword in kwargs:
+            raise ValueError(
+                keyword + ' argument not allowed, it will be overridden.')
+
+    kwargs['stdout'] = subprocess.PIPE
+    kwargs['stderr'] = subprocess.PIPE
+    kwargs['universal_newlines'] = True
+
+    process = subprocess.Popen(*args, **kwargs)
+    output, err = process.communicate()
+    retcode = process.poll()
+    if retcode:
+        cmd = kwargs.get('args')
+        if cmd is None:
+            cmd = args[0]
+        logger.debug(
+            "'%s' exited with %d. stdout output was:\n%s\nstderr output was:\n%s",
+            cmd, retcode, output, err)
+        raise subprocess.CalledProcessError(retcode, cmd)
+    return (output, err)
+
+
+def verify_exe_exists(exe, message=None):
+    """Ensures an executable with the given name is available.
+
+    If an executable isn't found for the given path or name, extra
+    directories are added to the user's PATH to help find system
+    utilities that may not be available in the default cron PATH.
+
+    :param str exe: executable path or name
+    :param str message: Error message to print.
+
+    :raises .NoInstallationError: when the executable isn't found
+
+    """
+    if message is None:
+        message = "Cannot find executable '{0}'.".format(exe)
+    if not (certbot_util.exe_exists(exe) or plugins_util.path_surgery(exe)):
+        raise errors.NoInstallationError(message)
+
+def report_master_overrides(name, overrides, acceptable_overrides=None):
```

Comment:
This functions behavior when `name` refers to TLS protocols is a little unusual. `acceptable_overrides` still needs to be set for it for it to consider other values, but `acceptable_overrides` isn't used.

Maybe we could refactor things a bit, however, I think we should at least add a comment describing this behavior.